libs/auth/src/common/services/pin/pin.service.implementation.ts

```diff
@@ -0,0 +1,420 @@
+import { firstValueFrom } from "rxjs";
+
+import { InternalMasterPasswordServiceAbstraction } from "@bitwarden/common/auth/abstractions/master-password.service.abstraction";
+import { KdfConfig } from "@bitwarden/common/auth/models/domain/kdf-config";
+import { EncryptService } from "@bitwarden/common/platform/abstractions/encrypt.service";
+import { KeyGenerationService } from "@bitwarden/common/platform/abstractions/key-generation.service";
+import { LogService } from "@bitwarden/common/platform/abstractions/log.service";
+import { StateService } from "@bitwarden/common/platform/abstractions/state.service";
+import { KdfType } from "@bitwarden/common/platform/enums";
+import { EncString, EncryptedString } from "@bitwarden/common/platform/models/domain/enc-string";
+import { SymmetricCryptoKey } from "@bitwarden/common/platform/models/domain/symmetric-crypto-key";
+import {
+  PIN_DISK,
+  PIN_MEMORY,
+  StateProvider,
+  UserKeyDefinition,
+} from "@bitwarden/common/platform/state";
+import { UserId } from "@bitwarden/common/types/guid";
+import { MasterKey, PinKey, UserKey } from "@bitwarden/common/types/key";
+
+import { PinServiceAbstraction } from "../../abstractions/pin.service.abstraction";
+
+/**
+ * - DISABLED   : No PIN set.
+ * - PERSISTENT : PIN is set and persists through client reset.
+ * - TRANSIENT  : PIN is set, but does not persist through client reset.
+ *                After client reset the master password is required to unlock.
+ */
+export type PinLockType = "DISABLED" | "PERSISTANT" | "TRANSIENT";
```

Comment:
⛏️ Looks like there's a typo here. If it's left in for backwards compatibility, just add a comment so that some later editor isn't tempted to fix it the wrong way.

```suggestion
export type PinLockType = "DISABLED" | "PERSISTENT" | "TRANSIENT";
```